client/src/search.tsx

```diff
@@ -240,10 +248,16 @@ function InnerSearchNavigateWidget(props: InnerSearchNavigateWidgetProps) {
       .toLowerCase()
       .split(" ")
       .map((s) => s.trim());
-    const indexResults: number[] = searchIndex.flex
+    const indexResults = searchIndex.flex
       .filter(([_, title]) => q.every((q) => title.includes(q)))
-      .map(([i]) => i)
+      .map(([index, title, slugTail]) => {
+        const exact = Number(q.length === 1 && q[0] === slugTail);
```

Comment:
I think exact matches should be by title, instead of by slug. For example, if I type "getting started", does it give me https://developer.mozilla.org/en-US/docs/Web/Guide/AJAX/Getting_Started instead of https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web?

Reply:
"getting started" would not be an exact match with "getting_started" for two reasons:
1. "getting started" has two terms, and I restricted exact matches to single-word queries.
2. "getting started" does not exactly match the slug tail "getting_started".

If we would only use an exact match on the title, you wouldn't find [`<head>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) by searching for "head".

Reply:
Maybe we can attempt both as exact matches? I certainly think matching by title is more intuitive than slugs for normal users since slug is kind of opaque.

Reply:
Do you have an example where the title matches exactly, but the slug does not?

Reply:
"Getting started" for example 😄 To get an exact match you would need to type "getting_started" but normal users don't search by slugs.

Reply:
Okay, I thought about it again, and I don't mind, even if this only boosts SVG's [Getting started](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Getting_Started) (Ajax's [Getting started](https://developer.mozilla.org/en-US/docs/Web/Guide/AJAX/Getting_Started) is already popular enough).

PS: Those pages should really be titled "Getting started with SVG" and "Getting started with AJAX" for consistency.

Reply:
> PS: Those pages should really be titled "Getting started with SVG" and "Getting started with AJAX" for consistency.

I agree—I was only using them to illustrate a potential UX issue.

Reply:
✅  in bc16acf9bb0ef1dd77f357f31af346f185a106f5.

Reply:
Thanks again for the valuable feedback, @Josh-Cena. 🙌